controllers/cloud.redhat.com/pool_controller.go

```diff
@@ -133,6 +134,12 @@ func (r *NamespacePoolReconciler) getPoolStatus(ctx context.Context, pool crd.Na
 				case "error":
```

Comment:
Solid point! I went ahead and updated the code here so that if a namespace is expired, it will delete the prometheus operator as well:
```
if r.Poller.namespaceIsExpired(expirationTS) {
    DeletePrometheusOperator(ctx, r.Client, res.Status.Namespace)  <---- Change is here

    if err := r.Client.Delete(ctx, &res); err != nil {
        r.Log.Error(err, "Unable to delete waiting reservation", "res-name", res.Name)
    } else {
	r.Log.Info("Reservation for namespace has expired while waiting. Deleting.", "res-name", res.Name)
    }
    return ctrl.Result{}, nil
}
```

Added a call to `DeletePrometheusOperator` and passed in the name of the namespace associated with the reservation as a parameter.